tests/suites/test_suite_psa_crypto.function

```diff
@@ -6813,6 +7294,481 @@ exit:
+
+    PSA_ASSERT( psa_sign_hash_start( &sign_operation, key, alg,
+                                     input_data->x, input_data->len ) );
+
+    PSA_ASSERT( psa_sign_hash_complete( &sign_operation, signature,
+                                        signature_size,
+                                        &signature_length ) );
+
+    PSA_ASSERT( psa_sign_hash_abort( &sign_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    psa_interruptible_set_max_ops( 1 );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    TEST_EQUAL( psa_verify_hash_complete( &verify_operation ),
+                PSA_OPERATION_INCOMPLETE );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    psa_interruptible_set_max_ops( PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    PSA_ASSERT( psa_verify_hash_complete( &verify_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+    /* --- Attempt double starts. --- */
+
+    PSA_ASSERT( psa_sign_hash_start( &sign_operation, key, alg,
+                                     input_data->x, input_data->len ) );
+
+    TEST_EQUAL( psa_sign_hash_start( &sign_operation, key, alg,
+                                     input_data->x, input_data->len ),
+                PSA_ERROR_BAD_STATE );
+
+    PSA_ASSERT( psa_sign_hash_abort( &sign_operation ) );
+
+    PSA_ASSERT( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ) );
+
+    TEST_EQUAL( psa_verify_hash_start( &verify_operation, key, alg,
+                                       input_data->x, input_data->len,
+                                       signature, signature_length ),
+                PSA_ERROR_BAD_STATE );
+
+    PSA_ASSERT( psa_verify_hash_abort( &verify_operation ) );
+
+exit:
+    /*
+     * Key attributes may have been returned by psa_get_key_attributes()
+     * thus reset them as required.
+     */
+    psa_reset_key_attributes( &attributes );
+
+    psa_destroy_key( key );
+    mbedtls_free( signature );
+    PSA_DONE( );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_ECP_RESTARTABLE */
+void interruptible_signverify_hash_negative_tests(int key_type_arg,
```

Comment:
Minor, non-blocker: I'm not sure "negative" is the best description here, as this function also tests things that are expected to succeed (like trashing the input buffer or changing max-ops mid-operation). Perhaps "edge cases"? Stress test? Unusual flows?

Reply:
Will do this with #7112 - its too small really to stand on its own.